src/core/metrics/app-metrics.ts

```diff
@@ -83,4 +117,55 @@ export class AppMetrics {
       }
     });
   }
+
+  watchList = {
+    // _done_userResourceAtom: null, // should be done before init
+    _done_currentEventResource: null,
+    _done_layersGlobalResource: null,
+    _done_analyticsResource: null,
+    // _done_areaLayersDetailsResourceAtom: null, // can be disable in url
+    _done_layersInAreaAndEventLayerResource: null,
+    // _done_eventListResource: null,
+    [EVENT_MAP_IDLE]: null,
+  };
+
+  watch(name: string) {
+    if (this.watchList[name] === null) {
+      const timing = performance.now();
+      this.watchList[name] = timing;
+      if (every(this.watchList, Boolean)) {
+        // last event should be EVENT_MAP_IDLE
+        if (name !== EVENT_MAP_IDLE) {
+          this.watchList[EVENT_MAP_IDLE] = null;
+          return;
+        }
```

Comment:
so if last event is not EVENT_MAP_IDLE we just don't send the request and don't do any actions about it?